Formula/opentsdb.rb

```diff
@@ -0,0 +1,151 @@
+class HbaseLZORequirement < Requirement
+  fatal true
+
+  satisfy(:build_env => false) { !Tab.for_name("hbase").without?("lzo") }
+
+  def message; <<-EOS.undent
+  hbase must not have disabled lzo compression to use it in opentsdb:
+    brew install hbase
+    not
+    brew install hbase --without-lzo
+  EOS
+  end
+end
+
+class Opentsdb < Formula
+  desc "Scalable, distributed Time Series Database."
+  homepage "http://opentsdb.net/"
+  url "https://github.com/OpenTSDB/opentsdb/releases/download/v2.2.0/opentsdb-2.2.0.tar.gz"
+  sha256 "5689d4d83ee21f1ce5892d064d6738bfa9fdef99f106f45d5c38eefb9476dfb5"
+
+  option "without-lzo", "Don't use LZO Compression"
+
+  if build.without? "lzo"
+    depends_on "hbase"
+  else
+    depends_on "hbase"
+    depends_on "lzo"
+    depends_on HbaseLZORequirement
+  end
```

Comment:
Please replace this `if`-`else` block with the following code snippet. The benefit is that there's less duplication and the dependencies can be more easily discovered by our dependency resolution code (`depends_on "lzo" => :recommended` is always preferable over `depends_on "lzo" if build.with?("lzo")`).

```ruby
depends_on "hbase"
depends_on "lzo" => :recommended
depends_on HbaseLZORequirement if build.with?("lzo")
```